src/background/user-storage.ts

```diff
@@ -2,13 +2,18 @@ import {DEFAULT_SETTINGS, DEFAULT_THEME} from '../defaults';
 import {debounce} from '../utils/debounce';
 import {isURLMatched} from '../utils/url';
 import {UserSettings} from '../definitions';
-import {readSyncStorage, readLocalStorage, writeSyncStorage, writeLocalStorage} from './utils/extension-api';
+import {readSyncStorage, readLocalStorage, writeSyncStorage, writeLocalStorage, subscribeToOuterSettingsChange} from './utils/extension-api';
 
 const SAVE_TIMEOUT = 1000;
 
+interface UserStorageOptions {
+    onRemoteSettingsChange: () => any;
+}
+
 export default class UserStorage {
-    constructor() {
+    constructor({onRemoteSettingsChange}: UserStorageOptions) {
         this.settings = null;
+        subscribeToOuterSettingsChange(this.loadSettings && onRemoteSettingsChange);
```

Comment:
Not sure if it will work now. `this.loadSettings &&` acts as boolean and will be always true. Did you mean something else? Don't forget that `this.loadSettings` is `async`.